provider/filestore.go

```diff
@@ -236,6 +237,15 @@ func (s3 S3FileStore) FilestoreType() pc.FileStoreType {
 	return S3
 }
 
+func (s3 S3FileStore) AddEnvVars(envVars map[string]string) map[string]string {
```

Comment:
I would make all of these pointer receivers, I know you don't need to but it makes creating interfaces simpler

Reply:
just as a side comment, we do this everywhere, we mix pointers and receivers. I don't know go well enough to understand the severity of this but should keep this in the backlog